File: Backend/app/db/models/lecture.py
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey, Index
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.db.base import Base


class Lecture(Base):
    __tablename__ = "lectures"

    id = Column(Integer, primary_key=True, index=True)
    course_id = Column(Integer, ForeignKey("courses.id"), nullable=False)
    name = Column(String(255), nullable=False)
    description = Column(Text, nullable=False)
    slug = Column(String(255), nullable=False, index=True)
    video_url = Column(String(500), nullable=False)
    created_at = Column(DateTime(timezone=True), server_default=func.now(), nullable=False)
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now(), nullable=False)
    deleted_at = Column(DateTime(timezone=True), nullable=True)

    # Relationships
    course = relationship("Course", back_populates="lectures")

    # Indexes for optimization
    __table_args__ = (
        Index('idx_lectures_course_id', 'course_id'),
        Index('idx_lectures_slug', 'slug'),
        Index('idx_lectures_deleted_at', 'deleted_at'),
        Index('idx_lectures_course_id_deleted_at', 'course_id', 'deleted_at'),
    ) 

File: Frontend/src/types/lecture.ts
/**
 * Tipos para la entidad Lecture (Clase)
 */

export interface Lecture {
  id: number;
  course_id: number;
  name: string;
  description: string;
  slug: string;
  video_url: string;
  created_at: string;
  updated_at: string;
  deleted_at: string | null;
}

export interface LectureCreate {
  course_id: number;
  name: string;
  description: string;
  slug: string;
  video_url: string;
}

export interface LectureUpdate {
  name?: string;
  description?: string;
  slug?: string;
  video_url?: string;
}

export interface LectureResponse {
  id: number;
  name: string;
  description: string;
  slug: string;
  video_url: string;
  created_at: string;
  updated_at: string;
}

export interface LectureListItem {
  id: number;
  name: string;
  description: string;
  slug: string;
}
